### PR TITLE
LPD-24678 Cannot export DDL list after remove a field from DDL definition

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
[LPD-24678](https://liferay.atlassian.net/browse/LPD-24678)

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/storage/DDMStorageEngineManager.java [hunk 1]
```diff
@@ -27,6 +28,9 @@ public long create(
 
 	public DDMFormValues getDDMFormValues(long classPK) throws PortalException;
 
+	public DDMFormValues getDDMFormValues(long classPK, DDMForm ddmForm)
+		throws PortalException;
+
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/storage/DDMStorageEngineManager.java:33]: @pedro-oliveira446 did you run `gw baseline`?

File: modules/apps/dynamic-data-lists/dynamic-data-lists-test/src/testIntegration/java/com/liferay/dynamic/data/lists/exporter/test/DDLExporterTest.java [hunk 2]
```diff
@@ -177,13 +177,20 @@ public void testExportRecordsWithDistinctFields() throws Exception {
 			DDMFormTestUtil.createTextDDMFormField(
 				"field0", false, false, false));
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-lists/dynamic-data-lists-test/src/testIntegration/java/com/liferay/dynamic/data/lists/exporter/test/DDLExporterTest.java:179]: @pedro-oliveira446 please remove the empty line if possible

File: modules/apps/dynamic-data-lists/dynamic-data-lists-test/src/testIntegration/java/com/liferay/dynamic/data/lists/exporter/test/DDLExporterTest.java [hunk 3]
```diff
@@ -177,13 +177,20 @@ public void testExportRecordsWithDistinctFields() throws Exception {
...
+			createDDMFormField("field1", "radio", "string"));
+
 		DDMFormValues ddmFormValues = DDMFormValuesTestUtil.createDDMFormValues(
 			ddmForm, _availableLocales, _defaultLocale);
 
 		ddmFormValues.addDDMFormFieldValue(
 			DDMFormValuesTestUtil.createDDMFormFieldValue(
 				"field0", new UnlocalizedValue("text0")));
 
+		ddmFormValues.addDDMFormFieldValue(
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-lists/dynamic-data-lists-test/src/testIntegration/java/com/liferay/dynamic/data/lists/exporter/test/DDLExporterTest.java:190]: @pedro-oliveira446 please remove the empty line if possible

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by carolmariaabb** [comment]
ci:test:sf

**Review by carolmariaabb — CHANGES_REQUESTED** [review]
@pedro-oliveira446 please fix the SF failure

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:forms

**Review by carolmariaabb — APPROVED** [review]

**Pull request closed without merging**